src/main/java/io/github/thebusybiscuit/slimefun4/utils/tags/SlimefunTag.java

```diff
@@ -0,0 +1,312 @@
+     */
+    CROP_GROWTH_ACCELERATOR_BLOCKS,
+
+    /**
+     * All <strong>strong</strong> materials which the {@link ClimbingPick} is able to climb.
+     */
+    CLIMBING_PICK_STRONG_SURFACES,
+
+    /**
+     * All <strong>weak</strong> materials which the {@link ClimbingPick} is able to climb.
+     */
+    CLIMBING_PICK_WEAK_SURFACES,
+
+    /**
+     * This {@link SlimefunTag} holds all surfaces for the {@link ClimbingPick}.
+     * This is an aggregation of {@code CLIMBING_PICK_STRONG_SURFACES} and {@code CLIMBING_PICK_WEAK_SURFACES}
+     */
+    CLIMBING_PICK_SURFACES,
+
+    /**
+     * All materials (ores) which trigger the Talisman of the Caveman.
+     */
+    CAVEMAN_TALISMAN_TRIGGERS;
+
+    private static final Map<String, SlimefunTag> nameLookup = new HashMap<>();
+    public static final SlimefunTag[] valuesCache = values();
+
+    static {
+        for (SlimefunTag tag : valuesCache) {
+            nameLookup.put(tag.name(), tag);
+        }
+    }
+
+    private final NamespacedKey key;
+    private final Set<Material> includedMaterials = new HashSet<>();
+    private final Set<Tag<Material>> additionalTags = new HashSet<>();
+
+    /**
+     * This constructs a new {@link SlimefunTag}.
+     * The {@link NamespacedKey} will be automatically inferred using
+     * {@link SlimefunPlugin} and {@link #name()}.
+     */
+    SlimefunTag() {
+        key = new NamespacedKey(SlimefunPlugin.instance(), name().toLowerCase(Locale.ROOT));
+    }
+
+    /**
+     * This method reloads this {@link SlimefunTag} from our resources directory.
+     * 
+     * @throws TagMisconfigurationException
+     *             This is thrown whenever a {@link SlimefunTag} could not be parsed properly
+     */
+    public void reload() throws TagMisconfigurationException {
+        new TagParser(this).parse(this, (materials, tags) -> {
+            this.includedMaterials.clear();
+            this.includedMaterials.addAll(materials);
+
+            this.additionalTags.clear();
+            this.additionalTags.addAll(tags);
+        });
+    }
+
+    public static void reloadAll() throws TagMisconfigurationException {
+        for (SlimefunTag tag : valuesCache) {
+            tag.reload();
+        }
+    }
+
+    @Nonnull
+    @Override
+    public NamespacedKey getKey() {
+        return key;
+    }
+
+    @Override
+    public boolean isTagged(@Nonnull Material item) {
+        if (includedMaterials.contains(item)) {
+            return true;
+        } else {
+            // Check if any of our additional Tags contain this Materials
+            for (Tag<Material> tag : additionalTags) {
+                if (tag.isTagged(item)) {
+                    return true;
+                }
+            }
+
+            // Now we can be sure it isn't tagged in any way
+            return false;
+        }
+    }
+
+    @Nonnull
+    @Override
+    public Set<Material> getValues() {
```

Comment:
This also could use some javadocs.

Reply:
It's an overridden method, it's just the implementation of the Tag#isTagged() method, the docs are inherited and would be redundant here.